src/plugins/data_views/common/data_views/data_view.ts

```diff
@@ -193,6 +194,22 @@ export class DataView extends AbstractDataView implements DataViewBase {
     return dataViewSpec;
   }
 
+  /**
+   * Can be used to check if two data views are the same in terms of their spec.
+   * Useful to detect changes. ES|QL typed data views don't take into account their id.
+   * The id is just used to be identified in the data view cache
+   * and where it's needed from consumers e.g. in Discover state
+   */
+  getSpecHash() {
+    const spec = this.toMinimalSpec();
+    const includeId = this.type !== ESQL_TYPE;
+    if (!includeId) {
+      // remove id from spec ES|QL data views
+      delete spec.id;
+    }
+    return JSON.stringify(spec);
+  }
```

Comment:
worth to note, that this could also use for general ad-hoc data views, fixing another problem, but let's focus on the first problem ... ES|QL

Reply:
So the hash is just the stringified minimal spec - we need to check and see what the limit is for the saved object id field.

I'm not sure this is useful for non ESQL data views since they could be compared based on identity due to the cache (object === object) or id.